source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
I see. I think that's still an independent issue that would be easiest to discuss in a separate pull request. Can you separate it into one, and copy paste your explanation and the error message? I don't think the issue is trivial.